Installation docs. First pass review and changes. (#283)
epinio/docs: docs/installation/install_epinio.md

:::info
To verify that your cluster provides a default StorageClass run the command `kubectl get storageclass`. The default StorageClass is marked with the string `(default)` next to its name in the output list.
:::

For example, you can deploy and configure the `local-path` dynamic storage provisioner by:
```bash
kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml
kubectl patch storageclass local-path -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
```

### Install Epinio

<!--TODO: I'm not sure about the structure/ordering of this section, down to "Verify Helm.....". Needs some thought-->

If the above dependencies are available or going to be installed by this chart,
Epinio can be installed with the following:

```bash
helm repo add epinio https://epinio.github.io/helm-charts
helm repo update
helm upgrade --install epinio epinio/epinio --namespace epinio --create-namespace \
    --set global.domain=myepiniodomain.org
```

<details>

<summary>Installation with "Let's Encrypt" certificates</summary>

> To generate trusted TLS certificates with "Let's Encrypt" for your public domain set `.Values.global.tlsIssuer` to `letsencrypt-production` and your e-mail to the value for the `.Values.global.tlsIssuerEmail` key.

Reviewer: ```suggestion
To generate trusted TLS certificates with "Let's Encrypt" for your public domain set `.Values.global.tlsIssuer` to `letsencrypt-production` and the value for the `.Values.global.tlsIssuerEmail` key to your e-mail.
```

Epinio can be installed with the following:

```bash
helm repo add epinio https://epinio.github.io/helm-charts
helm repo update
helm upgrade --install epinio epinio/epinio --namespace epinio --create-namespace \
    --set global.domain=myepiniodomain.org
```
Or you can install using "Let's Encrypt" certificates.

To generate trusted TLS certificates with "Let's Encrypt" for your public domain set `.Values.global.tlsIssuer` to `letsencrypt-production` and the value for the `.Values.global.tlsIssuerEmail` key to your e-mail address. Then:

```bash
helm repo add epinio https://epinio.github.io/helm-charts
helm repo update
 helm upgrade --install epinio epinio/epinio --namespace epinio --create-namespace \
    --set global.domain=myepiniodomain.org \
    --set global.tlsIssuer=letsencrypt-production \
    --set global.tlsIssuerEmail=user@company.org
```

The only mandatory field is the `.Values.global.domain` which should have the value of a wildcard `*.` enabled domain.
It should point to the IP address of your running Ingress controller.

:::tip

- To use a non-default IngressClass you need to specify it using `--set ingress.ingressClassName=<className>`

> - If you receive `Entity Too Large` errors when [uploading](https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#custom-max-body-size) application source code into Epinio, you need to increase `proxy-body-size` with `--set 'ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-body-size=1000m'`

Reviewer: ```suggestion
- If you receive `Entity Too Large` errors when [uploading](https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#custom-max-body-size) application source code into Epinio, you need to increase the `proxy-body-size` with `--set 'ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-body-size=1000m'`
```

nit.

       ghcr.io/epinio/<IMAGE>:v1.8.1
```

where `<IMAGE>` is the name of the image to verify.

## Installation on Specific Kubernetes Offerings

Installing Epinio, as described above, is a standard process, however you might need to configure it in a specific Kubernetes cluster.

To help you, see the following documents for some well-known clusters:

- [Install on Rancher](other_inst_scenarios/install_epinio_on_rancher.md) — Install Epinio on Rancher
- [Install on Public Cloud](other_inst_scenarios/install_epinio_on_public_cloud.md) — Install Epinio on Public Cloud cluster
- [Install on RKE2](other_inst_scenarios/install_epinio_on_rke.md) — Install Epinio on Rancher RKE2 cluster
- [Install on K3d](other_inst_scenarios/install_epinio_on_k3d.md) — Install Epinio on K3d cluster
- [Install on K3s](other_inst_scenarios/install_epinio_on_k3s.md) — Install Epinio on K3s cluster
- [Install on Rancher Desktop](other_inst_scenarios/install_epinio_on_rancher_desktop.md) — Install Epinio on Rancher Desktop
- [Install on EKS](other_inst_scenarios/install_epinio_on_eks.md) — Install Epinio on AWS EKS cluster

:::note

> The Public Cloud How-to describes the three major cloud providers but Epinio can run on any Kubernetes cluster.

Reviewer: Maybe link to this how to ?

Author: I will do.

:::

## Internal Epinio components

### Kubed

Kubed is installed as a subchart when `.Values.kubed.enabled` is `true` (default).

> If you already have kubed, you can skip installation by setting

Reviewer: ```suggestion
If you already have `kubed`, you can skip installation by setting
```

the helm value `.Values.kubed.enabled` to `false`.

### S3 storage

Epinio uses an S3 compatible storage to store the application source code.
This chart will install [Minio](https://min.io/) when `.Values.minio.enabled` is
`true` (default).

In addition to Minio, Epinio offers [s3gw](https://s3gw.io/) as another S3 compatible store.
It is installed when `.Values.minio.enabled` is set to `false` and `.Values.s3gw.enabled` is set to `true`.

Both choices for internal S3 compatible storage can be configured to use a user-defined storageClass.
If no StorageClass is defined, the default storageClass is used.
Using Minio set the custom storageClass by the value of `.Values.persistance.storageClass`.

> Using s3gw set the custom storageClass by the value of `.Values.s3gw.storageClass.name`.

Reviewer: ```suggestion
When using s3gw set the custom storageClass as the value of `.Values.s3gw.storageClass.name`.
```

(and minio equivalently)

> <!--TODO: Not sure about using a hashed blob, is that OK? Presumably to guarantee a line number reference?-->

Reviewer: Yes. A non-hashed reference, i.e. by branch name can shift what it points to, as the file changes on the branch.

Use any external S3 compatible solution by setting `.Values.minio.enabled` to `false` (`.Values.s3gw.enabled` is `false` by default) and using [the values under `s3`](https://github.com/epinio/helm-charts/blob/b389a4875af9f03b484a911c49a14f834ba04b64/chart/epinio/values.yaml#L44) to point to the required S3 server.

### Dex

[Dex](https://dexidp.io) OpenID Connect Provider is installed as a subchart when `.Values.global.dex.enabled` is set to `true` (default).

If you don't need to use an identity provider, set the value to `false` and use only local Epinio users. [OIDC Authentication](../references/authentication_oidc.md) has more information.

### Container Registry

When Epinio builds a container image for an application from source, it needs

> to store that image in container registry. Epinio installs a container registry

Reviewer: ```suggestion
to store that image in a container registry. Epinio installs a container registry
```
